nncf/torch/quantization/algo.py

```diff
@@ -1066,8 +1067,15 @@ def is_weights(ip: PTTargetPoint) -> bool:
                                             compression_lr_multiplier=compression_lr_multiplier)
         quantizer = self.__create_quantize_module(qspec).to(self._device_for_callable_obj_creation)
         if range_init_minmax_values is not None:
-            quantizer.apply_minmax_init(min_values=range_init_minmax_values[0],
-                                        max_values=range_init_minmax_values[1],
+            # Need to cast to the model's current dtype since the statistics could have been gathered in an
+            # AMP autocast model (and therefore be FP16 since AMP autocast switches precision of activations
+            # at forward pass time)
+            own_type = get_model_dtype(target_model)
```

Comment:
The `get_model_dtype` function returns the type of the first model parameter, am I right?

Reply:
Correct, this is basically the same way that we used to determine model device historically, only separated in a specific function.

Reply:
It looks like, It can be used for device determination. But I'm not sure that we can use the same way to determine the type of all model operations. Let's discuss it offline.